### ISSUE TITLE
`npm run watch` producing errors on a clean install? - Module build failed

### ISSUE THREAD

**Issue by 7immer** [nativescript-vue/vue-cli-template#28]
`npm run watch` producing errors on a clean install? I cannot successfully run the new project app due to this issue.

OS: MacOs
node: v9.5.0
npm: v5.6.0

ERROR in './src/components/HelloWorld.vue'
Module build failed: Error: 

Vue packages version mismatch:

- vue@2.4.2
- vue-template-compiler@2.5.16

This may cause things to work incorrectly. Make sure to use the same version for both.
If you are using vue-loader@>=10.0, simply update vue-template-compiler.
If you are using vue-loader@<10.0 or vueify, re-installing vue-loader/vueify should bump vue-template-compiler to the latest.

**Comment by rigor789** [comment]
Interesting, what commands did you use when creating the project?

**Comment by 7immer** [comment]
Hey @rigor789,
Thanks for looking into this :)

Commands entered are the same as the ones found here: https://nativescript-vue.netlify.com/en/docs/getting-started/quick-start/

```
$ npm install -g @vue/cli
$ npm install -g @vue/cli-init
$ vue init nativescript-vue/vue-cli-template tester
$ cd tester
$ npm install
$ npm run watch:ios
```

`tns doctor` says: No issues were detected.

iOS simulator starts, loads app and then it hangs at the splash screen.

Terminal history: https://pastebin.com/raw/9N4iN8y8

Hope this helps

**Comment by 7immer** [comment]
Is the installation and running process working for other people? Can I provide any more tests / data?

This template works fine on my machine: https://github.com/nativescript-vue/nativescript-vue-template.

**Comment by rigor789** [comment]
Yes it works for me just fine, i just ran the exact same commands as you did, and the app built, and ran properly. Here is my terminal history: 

```shell
⬢ Code ❯❯❯ vue init nativescript-vue/vue-cli-template tester

? Project name tester
? Project description A native application built with NativeScript-Vue
? Application name NativeScript-Vue Application
? Unique application identifier org.nativescript.application
? Project version 1.0.0
? Author rigor789 <rigor789@gmail.com>
? License MIT
? Install vue-router? (experimental) No
? Install vuex? (state management) No
? Color scheme light

   vue-cli · Generated "tester".

⬢ Code ❯❯❯ cd tester
⬢ tester ❯❯❯ npm i

> fsevents@1.1.3 install /Users/rigor789/Code/tester/node_modules/fsevents
> node install

[fsevents] Success: "/Users/rigor789/Code/tester/node_modules/fsevents/lib/binding/Release/node-v59-darwin-x64/fse.node" is installed via remote

> node-sass@4.8.3 install /Users/rigor789/Code/tester/node_modules/node-sass
> node scripts/install.js

Downloading binary from https://github.com/sass/node-sass/releases/download/v4.8.3/darwin-x64-59_binding.node
Download complete  ⸩ ⠋ :
Binary saved to /Users/rigor789/Code/tester/node_modules/node-sass/vendor/darwin-x64-59/binding.node
Caching binary to /Users/rigor789/.npm/node-sass/4.8.3/darwin-x64-59_binding.node

> uglifyjs-webpack-plugin@0.4.6 postinstall /Users/rigor789/Code/tester/node_modules/uglifyjs-webpack-plugin
> node lib/post_install.js


> node-sass@4.8.3 postinstall /Users/rigor789/Code/tester/node_modules/node-sass
> node scripts/build.js

Binary found at /Users/rigor789/Code/tester/node_modules/node-sass/vendor/darwin-x64-59/binding.node
Testing binary
Binary is fine
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN ajv-keywords@3.1.0 requires a peer of ajv@^6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN tester@1.0.0 No repository field.

added 914 packages from 574 contributors in 36.317s
⬢ tester ❯❯❯ npm run watch:ios

> tester@1.0.0 watch:ios /Users/rigor789/Code/tester
> npm run watch -- --env.ios


> tester@1.0.0 watch /Users/rigor789/Code/tester
> webpack --watch --env.tnsAction run "--env.ios"

info: Copying NativeScript plugins to template dependencies...
info: Preparing NativeScript application from template...
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN tester@ No repository field.

info: Bundling application for ios...

Webpack is watching the files…

Executing post-build scripts
info: Running NativeScript application...
Searching for devices...
Copying template files...
Installing tns-ios
Installing  tns-ios
+ tns-ios@3.4.1
added 1 package from 1 contributor in 2.235s
Project successfully created.
Preparing project...
Successfully prepared plugin nativescript-theme-core for ios.
Successfully prepared plugin nativescript-vue for ios.
Successfully prepared plugin tns-core-modules for ios.
Successfully prepared plugin tns-core-modules-widgets for ios.
Project successfully prepared (iOS)
Building project...
Xcode build...
=== BUILD TARGET dist OF PROJECT dist WITH CONFIGURATION Debug ===
while processing /Users/rigor789/Code/tester/dist/platforms/ios/internal/TKLiveSync/TKLiveSync(TKLiveSync.o)while processing /Users/rigor789/Code/tester/dist/platforms/ios/internal/TKLiveSync/TKLiveSync(TKLiveSync.o):
warning: /Users/nsbuilduser/Library/Developer/Xcode/DerivedData/ModuleCache/28BJBFDGVGZZ3/UIKit-1V5UHAPTOD24G.pcm: No such file or directory
note: Linking a static library that was built with -gmodules, but the module cache was not found.  Redistributable static libraries should never be built with module debugging enabled.  The debug experience will be degraded due to incomplete debug information.
:
warning: /Users/nsbuilduser/Library/Developer/Xcode/DerivedData/ModuleCache/2DKXPQ92SAISO/UIKit-1V5UHAPTOD24G.pcm: No such file or directory
while processing /Users/rigor789/Code/tester/dist/platforms/ios/internal/TKLiveSync/TKLiveSync(TKLiveSync.o)note: Linking a static library that was built with -gmodules, but the module cache was not found.  Redistributable static libraries should never be built with module debugging enabled.  The debug experience will be degraded due to incomplete debug information.
:
warning: /Users/nsbuilduser/Library/Developer/Xcode/DerivedData/ModuleCache/28BJBFDGVGZZ3/Darwin-38I6DLZ5IH61J.pcm: No such file or directory
while processing /Users/rigor789/Code/tester/dist/platforms/ios/internal/TKLiveSync/TKLiveSync(TKLiveSync.o):
warning: /Users/nsbuilduser/Library/Developer/Xcode/DerivedData/ModuleCache/2DKXPQ92SAISO/Darwin-38I6DLZ5IH61J.pcm: No such file or directory
while processing /Users/rigor789/Code/tester/dist/platforms/ios/internal/TKLiveSync/TKLiveSync(TKLiveSync.o):
warning: /Users/nsbuilduser/Library/Developer/Xcode/DerivedData/ModuleCache/28BJBFDGVGZZ3/Foundation-3DFYNEBRQSXST.pcm: while processing No such file or directory
/Users/rigor789/Code/tester/dist/platforms/ios/internal/TKLiveSync/TKLiveSync(TKLiveSync.o):
warning: /Users/nsbuilduser/Library/Developer/Xcode/DerivedData/ModuleCache/2DKXPQ92SAISO/Foundation-3DFYNEBRQSXST.pcm: No such file or directorywhile processing /Users/rigor789/Code/tester/dist/platforms/ios/internal/TKLiveSync/TKLiveSync(TKLiveSync.o):

warning: /Users/nsbuilduser/Library/Developer/Xcode/DerivedData/ModuleCache/28BJBFDGVGZZ3/Dispatch-2LX9MWM6UEPQW.pcm: No such file or directory
while processing while processing /Users/rigor789/Code/tester/dist/platforms/ios/internal/TKLiveSync/TKLiveSync(TKLiveSync.o)/Users/rigor789/Code/tester/dist/platforms/ios/internal/TKLiveSync/TKLiveSync(TKLiveSync.o):
:
warning: warning: /Users/nsbuilduser/Library/Developer/Xcode/DerivedData/ModuleCache/2DKXPQ92SAISO/Dispatch-2LX9MWM6UEPQW.pcm/Users/nsbuilduser/Library/Developer/Xcode/DerivedData/ModuleCache/28BJBFDGVGZZ3/ObjectiveC-1RQA2RIXJIE19.pcm: : No such file or directoryNo such file or directory

while processing /Users/rigor789/Code/tester/dist/platforms/ios/internal/TKLiveSync/TKLiveSync(TKLiveSync.o):
warning: /Users/nsbuilduser/Library/Developer/Xcode/DerivedData/ModuleCache/2DKXPQ92SAISO/ObjectiveC-1RQA2RIXJIE19.pcm: No such file or directory
Project successfully built.
Installing...
Successfully installed on device with identifier '0EC95A2B-EF10-4BA3-BCC4-181C46895002'.
Successfully transferred all files.
Refreshing application...
Successfully synced application org.nativescript.application on device 0EC95A2B-EF10-4BA3-BCC4-181C46895002.
CONSOLE LOG file:///app/app.js:1:93893: 'NativeScript-Vue has "Vue.config.silent" set to true, to see output logs set it to false.'
^C
```

**Comment by 7immer** [comment]
I've got it working! 

The trick was to `npm uninstall vue` and then `npm install vue`
Alternatively before running npm install on a clean template, first add `"vue": "^2.5.16"` to `packages.json` `dependences`.

The error and the above solution is reproducible on my machine. (ie I have to do the above for each new template)

--

Before getting to the above solution I also tried: removing `~/.vue-template`, `npm clear cache`, updating node and npm via nvm to the latest version, reinstalling `nativescript`, `vue-cli`, `vue-init` and many of the solutions here: https://github.com/vuejs/vue/issues/3941.
